gluecodium/src/test/resources/smoke/generic_types/output/android/jni/com_example_smoke_GenericTypesWithGenericTypes.cpp

```diff
@@ -115,6 +116,8 @@ Java_com_example_smoke_GenericTypesWithGenericTypes_methodWithMapGenericKeys(JNI
 JNIEXPORT void JNICALL
 Java_com_example_smoke_GenericTypesWithGenericTypes_disposeNativeHandle(JNIEnv* _jenv, jobject _jinstance, jlong _jpointerRef)
 {
-    delete reinterpret_cast<std::shared_ptr<::smoke::GenericTypesWithGenericTypes>*> (_jpointerRef);
+    auto p_nobj = reinterpret_cast<std::shared_ptr<::smoke::GenericTypesWithGenericTypes>*>(_jpointerRef);
```

Comment:
I would protect such code with unique_ptr

Reply:
Seems unnecessary as this is the only code path that could delete the shared_ptr.